src/util/util_yt_download.py

```diff
@@ -0,0 +1,117 @@
+        """Downloads the audio of the given url. Should not be called from outside, use download() instead.
+
+        Args:
+            url: Youtube url of the video.
+        """
+        if self.is_url_downloaded_single(self, url):
+            logging.info(f"Url {url} is already downloaded")
+        else:
+            logging.info(f"Started download for url {url}")
+            video = YouTube(url)
+            filename = video.video_id + "." + FILE_EXTENSION
+            video.streams.filter(
+                file_extension=FILE_EXTENSION, only_audio=True
+            ).first().download(output_path=OUTPUT_PATH, filename=filename)
+
+            YOUTUBE_TABLE.insert(
+                url = url,
+                title = video.title,
+                length = video.length,
+                filepath = OUTPUT_PATH,
+                filename = filename,
+                thumbnail_url = video.thumbnail_url,
+                timestamp = int(time.time()),
+                hash_ = self.compute_hash(self, OUTPUT_PATH + filename),
+            )
+
+            logging.info(f"Finished download for url {url}")
+            logging.info(f"Saved file in {OUTPUT_PATH + filename}")
+
+    async def __download_playlist(self, url):
+        """Downloads the audio of the given playlist url. Should not be called from outside, use download() instead.
+
+        Args:
+            url: Youtube url of the playlist.
+        """
+        logging.info(f"Started download for playlist url {url}")
+        playlist = Playlist(url)
+        logging.info(f"Number of videos in playlist: {playlist.length}")
+
+        index = 0
+        for video_url in playlist.video_urls:
+            index += 1
+            await self.__download_single(self, video_url)
+            logging.info(f"Finished download for video nr. {index}")
+
+        logging.info(f"Finished download for playlist url {url}")
+
+    async def download(self, url):
+        """Downloads the audio of the given url.
+
+        Args:
+            url: Youtube url of the video.
+        """
+        if "list" in url:
+            self.__download_playlist(self, url)
+        else:
+            self.__download_single(self, url)
+
+    async def is_url_downloaded_single(self, url):
+        """Checks if the given url is already downloaded.
+
+        Args:
+            url: Youtube url of the video.
+
+        Returns:
+            bool: true - url already downloaded
+        """
+        logging.info(f"Checking url {url}")
+
+        if YOUTUBE_TABLE.get_or_none(YOUTUBE_TABLE.url == url):
+            return False
+
+        return True
+    
+    async def compute_hash(self, filepath):
+        """Compute the MD5Hash of a given mp4 file.
+
+        Args:
+            filepath: Filepath of the mp4 file.
+
+        Returns:
+            string: readable_hash - MD5Hash of the given mp4 file
+        """
+        logging.info(f"Computing hash for file {filepath}")
+
+        md5_hash = hashlib.md5()
```

Comment:
Please use SHA256. MD5 is broken (https://en.wikipedia.org/wiki/MD5#Security).

I know, it shouldn't matter here since it's only intended for checking file corruption and no secure transfers, but it's easy to change to SHA256, so why not.

Reply:
It does absolutly not matter in this context, but sure let's go wit sha.